addons/mrp/models/mrp_bom.py

```diff
@@ -114,6 +114,14 @@ def name_create(self, name):
     def name_get(self):
         return [(bom.id, '%s%s' % (bom.code and '%s: ' % bom.code or '', bom.product_tmpl_id.display_name)) for bom in self]
 
+    @api.constrains('product_tmpl_id', 'product_id', 'type')
+    def check_kit_has_not_orderpoint(self):
+        for bom in self.filtered(lambda bom: bom.type == "phantom"):
+            product_ids = bom.product_id if bom.product_id \
```

Comment:
To be more concise:
`product_ids = bom.product_id or bom.product_tmpl_id.product_variant_ids`